src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -263,7 +263,9 @@
                                        [:= :certnames.certname :fs.certname]
 
                                        :reports
-                                       [:= :certnames.latest_report_id :reports.id]
+                                       [:and
+                                        [:= :certnames.certname :reports.certname]
+                                        [:= :certnames.latest_report_id :reports.id]]
```

Comment:
@mullr in addition to here should we be adding the certname join criteria in [latest-report-query](https://github.com/puppetlabs/puppetdb/pull/2293/files#diff-fefe75691a2bbea085773295ea2b7c49R940) as well?